packages/ckb-sdk-core/src/index.ts

```diff
@@ -218,8 +262,202 @@ class Core {
       safeMode,
       cells: availableCells,
       deps,
+      capacityThreshold,
+      changeThreshold,
     })
   }
+
+  public generateDaoDepositTransaction = async ({
+    fromAddress,
+    capacity,
+    fee,
+    cells = [],
+  }: {
+    fromAddress: string,
+    capacity: bigint,
+    fee: bigint,
+    cells?: CachedCell[],
+  }) => {
+    if (!this.config.daoDep) {
+      await this.loadDaoDep()
+    }
+
+    if (!this.config.secp256k1Dep) {
+      await this.loadSecp256k1Dep()
+    }
+
+    const rawTx = await this.generateRawTransaction({
+      fromAddress,
+      toAddress: fromAddress,
+      capacity,
+      fee,
+      safeMode: true,
+      cells,
+      deps: this.config.secp256k1Dep!,
+    })
+
+
+    rawTx.outputs[0].type = {
+      codeHash: this.config.daoDep!.typeHash!,
+      args: '0x',
+      hashType: this.config.daoDep!.hashType,
+    }
+
+    rawTx.outputsData[0] = '0x0000000000000000'
+
+    rawTx.cellDeps.push({
+      outPoint: this.config.daoDep!.outPoint,
+      depType: 'code',
+    })
+    rawTx.witnesses.unshift({ lock: '', inputType: '', outputType: '' })
+
+    return rawTx
+  }
+
+  public generateDaoWithdrawStartTransaction = async ({ outPoint, fee, cells = [] }: {
+    outPoint: CKBComponents.OutPoint,
+    fee: bigint | string,
+    cells?: CachedCell[]
+  }) => {
+    if (!this.config.secp256k1Dep) {
+      await this.loadSecp256k1Dep()
+    }
+    if (!this.config.daoDep) {
+      await this.loadDaoDep()
+    }
+
+    const cellStatus = await this.rpc.getLiveCell(outPoint, false)
+    if (cellStatus.status !== 'live') throw new Error('Cell is not live yet.')
+
+    const tx = await this.rpc.getTransaction(outPoint.txHash)
+    if (tx.txStatus.blockHash === null) throw new Error('The cell is not a NervosDAO cell')
```

Comment:
?

Reply:
The block hash is possible to be `null` as the RPC docs say https://github.com/nervosnetwork/ckb/tree/develop/rpc#get_transaction

Reply:
`block_hash = null` means not committed, why you consider its not a NervosDao cell ?

Reply:
Removed in https://github.com/nervosnetwork/ckb-sdk-js/pull/382/commits/c88a8a5f15a4d093cf8e393f19d727c8cbc0611f